### PR TITLE
Feature/showevent

### PR DESCRIPTION
**Pull request by nelliei**
Showing upcoming events on the profile page

### CODE REVIEW

**Comment by ivarshav** [comment]
please resolve conflicts :)

**Comment by ivarshav** [comment]
please resolve conflicts :)

File: app/templates/profile .html [hunk 1]
```diff
@@ -0,0 +1,348 @@
+z{% extends "base.html" %} {% include "event/partials/text_editor_partial_head.html" %} {% block content %}
```
Comment (yammesicka) [inline app/templates/profile .html]: This file is `profile[SPACE].html`, I think you can delete it :)

File: app/templates/partials/user_profile/middle_content/event_card.html [hunk 2]
```diff
@@ -1,15 +1,15 @@
 <div class="card card-event mb-2 pb-0">
   <div class="card-header d-flex justify-content-between">
-    <small><i>{{ gettext("Upcoming event on (date)", date=event.start) }}</i></small>
+    <small><i>Upcoming event on {{ event.start.strftime("%d/%m/%Y %H:%M") }} </i></small>
```
Comment (yammesicka) [inline app/templates/partials/user_profile/middle_content/event_card.html]: Remove trailing space

File: app/internal/showevent.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    temp = events.copy()
+    return sorted(temp, key=attrgetter('start'))
+
+
+def filter_by_now(events: List[Event]) -> List[Event]:
+    upcoming_events = []
+    for event in events:
+        if event.start >= datetime.now():
+            upcoming_events.append(event)
+    return upcoming_events
```
Comment (yammesicka) [inline app/internal/showevent.py]: This all should be in one query :)

**Review by yammesicka — CHANGES_REQUESTED** [review]
Great work on the HTMLs. Now let's refactor some of the Python code to be even better :)

File: app/templates/profile .html (hunk 1 above)
Comment (nelliei) [reply app/templates/profile .html]: done :)

File: app/templates/partials/user_profile/middle_content/event_card.html (hunk 2 above)
Comment (nelliei) [reply app/templates/partials/user_profile/middle_content/event_card.html]: done

File: app/internal/showevent.py (hunk 3 above)
Comment (nelliei) [reply app/internal/showevent.py]: done :)

File: app/routers/profile.py [hunk 4]
```diff
@@ -41,14 +42,12 @@ async def profile(
...
         new_user=Depends(get_placeholder_user)):
-    # Get relevant data from database
-    upcoming_events = range(5)
     user = session.query(User).filter_by(id=1).first()
     if not user:
         session.add(new_user)
         session.commit()
         user = session.query(User).filter_by(id=1).first()
-
+    upcoming_events = get_upcoming_events(session, user.id)[:5]
```
Comment (ivarshav) [inline app/routers/profile.py]: what is 5?

Comment (nelliei) [reply app/routers/profile.py]: I added a constant variable and an explanation to it.
We are presenting up to five upcoming events on the profile page

File: app/routers/profile.py [hunk 5]
```diff
@@ -14,9 +14,12 @@
 from app.internal.import_holidays import (get_holidays_from_file,
                                           save_holidays_to_db)
 from app.internal.privacy import PrivacyKinds
+from app.internal.showevent import get_upcoming_events
 
 PICTURE_EXTENSION = config.PICTURE_EXTENSION
 PICTURE_SIZE = config.AVATAR_SIZE
+FIVE_EVENTS = 5
+"""We are presenting up to five upcoming events on the profile page"""
```
Comment (Gonzom) [inline app/routers/profile.py]: inline comment should not be triple quote but with "#".

Comment (nelliei) [reply app/routers/profile.py]: done

File: app/internal/showevent.py [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+from app.database.models import Event, UserEvent
+
+
+def get_upcoming_events(session: Session, user_id: int) -> List[Event]:
+    upcoming_events = (
+        session.query(Event).join(UserEvent)
+        .filter(UserEvent.user_id == user_id)
+        .filter(Event.start >= datetime.now())
+        .order_by(Event.start)
+        )
```
Comment (yammesicka) [inline app/internal/showevent.py]: precommit hooks should fix this over indentation

File: app/templates/partials/user_profile/middle_content.html [hunk 7]
```diff
@@ -3,6 +3,7 @@
   </div>
   <!-- Upcoming events -->
   <div class="mb-3">
+    <h3>Upcoming events</h3>
```
Comment (yammesicka) [inline app/templates/partials/user_profile/middle_content.html]: Are there h1 and h2 before? If not - use the highest unused title.
To reach a certain visual size, use CSS.

File: app/templates/partials/user_profile/middle_content/event_card.html [hunk 8]
```diff
@@ -1,15 +1,15 @@
...
   </div>
 
   <div class="card-body pb-1">
     <p class="card-text">
-      The Event {{ event }} - description ...
+      {{ event.title }} - {{ event.content if event.content is not none }}
     </p>
     <hr class="mb-0">
-    <small class="event-posted-time text-muted">Last updated {{time}} ago</small>
+    <a href="/event/{{ event.id }}">{{ event.title }} details</a>
```
Comment (yammesicka) [inline app/templates/partials/user_profile/middle_content/event_card.html]: Prefer `url_for`

**Review by yammesicka — CHANGES_REQUESTED** [review]
Great job! I've also added few insights, please take a look

File: app/internal/showevent.py (hunk 6 above)
Comment (nelliei) [reply app/internal/showevent.py]: Precommit hooks installed
Done

File: app/templates/partials/user_profile/middle_content/event_card.html (hunk 8 above)
Comment (nelliei) [reply app/templates/partials/user_profile/middle_content/event_card.html]: done

File: app/templates/partials/user_profile/middle_content.html (hunk 7 above)
Comment (nelliei) [reply app/templates/partials/user_profile/middle_content.html]: done

**Pull request merged**